dss/operations/util.py

```diff
@@ -39,3 +44,137 @@ def map_bucket_results(func: typing.Callable, handle: BlobStore, bucket: str, ba
+    Args:
+        environment: A dictionary containing all environment variables.
+        env_set_fn: A function handle used to set a new environment.
+        env_var: The name of the environment variable to unset.
+        where: A label for telling the user where the env var was unset.
+    """
+    try:
+        del environment[env_var]
+        env_set_fn(environment)
+        print(f'Unset variable "{env_var}" {where}')
+    except KeyError:
+        print(f'Nothing to unset for variable "{env_var}" {where}')
+
+def get_cloud_variable_prefix() -> str:
+    """
+    Use information from the environment to assemble
+    the necessary prefix for accessing variables in
+    the secrets manager or paramter store.
+    """
+    store_name = os.environ["DSS_PARAMETER_STORE"]
+    stage_name = os.environ["DSS_DEPLOYMENT_STAGE"]
+    store_prefix = f"{store_name}/{stage_name}"
+    return store_prefix
+
+def fix_cloud_variable_prefix(secret_name: str) -> str:
+    """
+    This adds the variable store and stage prefix
+    to the front of a variable name.
+    """
+    prefix = get_cloud_variable_prefix()
+    if not secret_name.startswith(prefix):
+        secret_name = f"{prefix}/{secret_name}"
+    return secret_name
+
+
+def get_elasticsearch_endpoint() -> str:
+    domain_name = os.environ['DSS_ES_DOMAIN']
+    domain_info = es_client.describe_elasticsearch_domain(DomainName=domain_name)
+    return domain_info['DomainStatus']['Endpoint']
+
+
+def get_admin_emails() -> str:
+    gcp_var = 'GOOGLE_APPLICATION_CREDENTIALS_SECRETS_NAME'
+    gcp_secret_id = fix_cloud_variable_prefix(gcp_var)
+
+    admin_var = 'ADMIN_USER_EMAILS_SECRETS_NAME'
+    admin_secret_id = fix_cloud_variable_prefix(admin_var)
+
+    if secret_is_gettable(gcp_secret_id) and secret_is_gettable(admin_secret_id):
+        resp = sm_client.get_secret_value(gcp_secret_id)
+        gcp_service_account_email = json.loads(resp['SecretString'])['client_email']
+        resp = sm_client.get_secret_value(admin_secret_id)
+        email_list = [email for email in resp['SecretString'].split(',') if email.strip()]
+        if gcp_service_account_email not in email_list:
+            email_list.append(gcp_service_account_email)
+        return ",".join(email_list)
+    else:
+        err = "Error adding Google service account email to admin emails list: "
+        if secret_is_gettable(gcp_secret_id):
+            err += f"Could not get secret {admin_var}"
+            raise RuntimeError(err)
+        elif secret_is_gettable(admin_secret_id):
+            err += f"Could not get secret {gcp_secret_id}"
+            raise RuntimeError(err)
+        else:
+            err += f"Unknown error occurred"
+            raise RuntimeError(err)
+
+
+# Secrets can be in three different states:
+# - secret exists (gettable, settable)
+# - secret exists but is marked for deletion (not gettable, not settable)
+# - secret does not exist (not gettable, settable)
+
+def secret_is_gettable(secret_name):
```

Comment:
Please move this method into `dss/operations/secrets.py`.

Reply:
Done: https://github.com/HumanCellAtlas/data-store/blob/dss-params/dss/operations/secrets.py#L25